tools/devtool

```diff
@@ -449,6 +449,11 @@ cmd_build() {
     # Cargo uses the debug profile by default. If we're building the release
     # binaries, we need to pass an extra argument to cargo.
     cargo_args=("$@")
+    
+    # Add the default target if we did not get that argument in the build command.
+    if [[ ! "${@}" =~ "--target" ]]; then
```

Comment:
We also need to update the `friendly_bin_dir` because no matter what you specify as the target, the script will place the binaries under `debug-musl` or `release-musl`.

I think it is a bit complicated to do find a truly friendly path for the binaries so I would suggest leaving the binaries in their default path in this case because `cargo` will probably do a better job than we can. One reason for which the fix in this PR is needed is for the ability to build aarch64 binaries on x86. In this case, you don't want the aarch64 binaries to override the x86 ones. So when building the "friendly" path, you would need to also take this into consideration. I think that it would complicate the code and I don't find it necessary, but @dhrgit should also have an input on this.

Reply:
Yes, I believe we can get rid of `$friendly_bin_dir`, since that was since before we had so many build options.

Coming back to this solution though, regex matching the expanded array of args will yield an unwanted match if "--target" is part of the command line, but not an arg name. E.g. it will still match a command line like `--some-key="this is not the --target arg, but some value"`.

Reply:
So from my understanding, we want to let cargo place the binaries where it will and print that path out after a build succeeds? I can adjust the that section if so.

@dhrgit With respect to your comment, I thought about iterating through the arguments and looking for an exact match initially. Would this be a better option here?

@andreeaflorescu I'll adjust my commit and commit message accordingly.

Reply:
> So from my understanding, we want to let cargo place the binaries where it will and print that path out after a build succeeds? I can adjust the that section if so.

Yap. We should first check if the CI counts on the binaries being placed in $friendly_bin_dir.

Reply:
@andreeaflorescu It's fine, nothing relies on the friendly bin path - it was just there to bring the binaries to a more easily user accessible location.

@gsquire Yes, I believe that, in order to detect a `--target` arg passed explicitly by the user, we'd need to inspect all args passed after the first `--` and before a next potential `--`. To complicate things further, key-values pairs can use either one (`--target=TARGET`) or two args (`--target TARGET`).

Reply:
I removed the `friendly_bin_dir` and added a more robust target flag check. I kept a pattern match for the `--target=...` possibility. Hopefully this is what you had in mind.

Thanks for the reviews.